src/nnet3/nnet-optimize.h

```diff
@@ -76,29 +77,50 @@ void Optimize(const NnetOptimizeOptions &config,
               const ComputationRequest &request,
               NnetComputation *computation);
 
-
+struct ComputationRequestHasher {
+  size_t operator()(const ComputationRequest *cr) const {
```

Comment:
This hashing function is very insufficient.
The input.size() and outputs.size() will normally be 1, so everything will hash to the same value.  You should go right in and look at the individual indexes, plus the names of the inputs and outputs.